packages/orbit-components/src/Separator/index.tsx

```diff
@@ -1,77 +1,79 @@
+import { getSpaceAfterClasses } from "../common/tailwind";
 
-function capitalize(string: string) {
-  return string.charAt(0).toUpperCase() + string.slice(1);
+enum BorderTypes {
+  none = "border-none",
+  solid = "border-solid",
+  dashed = "border-dashed",
+  dotted = "border-dotted",
+  double = "border-double",
 }
 
-function getIndentAmount({
-  theme,
-  $indent,
-}: // StyledContainer takes more props, so this shouldn't be exact
-{
-  theme: Theme;
-  $indent: Indent;
-}) {
-  return $indent === "none" ? 0 : theme.orbit[`space${capitalize($indent)}`];
+enum Indent {
+  none = "none",
+  small = "small",
+  medium = "medium",
+  large = "large",
+  XLarge = "XLarge",
+  XXLarge = "XXLarge",
 }
 
-const StyledContainer = styled.div<{ $align: Align; $indent: Indent }>`
-  ${({ $align }) => css`
-    box-sizing: border-box;
-    width: 100%;
-    padding-${right}: ${$align === "left" && getIndentAmount};
-    padding-${left}: ${$align === "right" && getIndentAmount};
-    padding: ${$align === "center" && `0 ${getIndentAmount}`};
-  `};
-`;
+function getIndentAmount(indent: keyof typeof Indent, align: Align) {
+  const classes = {
+    left: {
+      [Indent.none]: "0",
+      [Indent.small]: "pe-sm",
+      [Indent.medium]: "pe-md",
+      [Indent.large]: "pe-lg",
+      [Indent.XLarge]: "pe-xl",
+      [Indent.XXLarge]: "pe-xxl",
+    },
+    right: {
+      [Indent.none]: "0",
+      [Indent.small]: "ps-sm",
+      [Indent.medium]: "ps-md",
+      [Indent.large]: "ps-lg",
+      [Indent.XLarge]: "ps-xl",
+      [Indent.XXLarge]: "ps-xxl",
+    },
+    center: {
+      [Indent.none]: "0",
+      [Indent.small]: "px-sm",
+      [Indent.medium]: "px-md",
+      [Indent.large]: "px-lg",
+      [Indent.XLarge]: "px-xl",
+      [Indent.XXLarge]: "px-xxl",
+    },
+  };
 
-export const StyledSeparator = styled.hr<{
-  spaceAfter?: Common.SpaceAfterSizes;
-  $type: Props["type"];
-  $color?: Props["color"];
-}>`
-  ${({ theme, $type, $color }) => css`
-    height: ${theme.orbit.heightSeparator};
-    background: ${!$color && theme.orbit.backgroundSeparator};
-    box-sizing: border-box;
-    border-style: ${$type};
-    border-color: ${$color};
-    margin-top: 0;
-    margin-bottom: ${getSpacingToken};
-  `};
-`;
-
-StyledSeparator.defaultProps = {
-  theme: defaultTheme,
-};
+  return classes[align][indent];
+}
 
 const Separator = ({
   align = "left",
   indent = "none",
   spaceAfter,
-  type = "none",
-  color,
+  type = "solid",
+  borderColor,
 }: Props) => {
-  const theme = useTheme();
   return (
-    <StyledContainer $align={align} $indent={indent}>
-      <StyledSeparator
-        $type={type}
-        spaceAfter={spaceAfter}
-        $color={color && (theme.orbit[color] as Props["color"])}
+    <div className={cx("box-border w-full", getIndentAmount(indent, align))}>
+      <hr
+        className={cx(
+          "orbit-separator",
+          // h-0 and border-t are the same as in default tailwind preflight
```

Comment:
I'd say we don't need this comment. I'm ok with relying on preflight. I am concerned about how it can affect who is not using it, though. Nothing to change here, just so we are aware.

Reply:
actually, it's the opposite, by adding this we aren't relying on preflight, because I feel that in case if they won't use it -> the component will be broken, so I explicitly added it here :)